sdk/core/Azure.Core/src/ResponseClassifier.cs

```diff
@@ -2,15 +2,51 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections.Generic;
+using System.Globalization;
 using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
 
 namespace Azure.Core
 {
     /// <summary>
-    /// A type that analyzes HTTP responses and exceptions and determines if they should be retried.
+    /// This type has two responsibilities:
+    ///  1) Analyzes HTTP responses and exceptions and determine if they should be retried, or classified as an error
+    ///  2) Create the exception message in a service-specific way, with two customization options:
+    ///     A) Services may pass client options to customize how the error message is sanitized
+    ///     B) Services may inherit from ResponseClassifier and override ExtractFailureContent to customize how error
+    ///        information is extracted from the Response body.
     /// </summary>
     public class ResponseClassifier
     {
+        private const string DefaultMessage = "Service request failed.";
+
+        private readonly HttpMessageSanitizer _sanitizer;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ResponseClassifier"/>.
+        /// </summary>
+        public ResponseClassifier() : this(new DefaultClientOptions())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ResponseClassifier"/>.
+        /// </summary>
+        /// <param name="options"></param>
+        public ResponseClassifier(ClientOptions options)
```

Comment:
This is interesting. We will need to enforce everyone using this ctor and ban the parameterless one.

Reply:
I don't know that they always can -- sometimes ResponseClassifier is newed in Core and would take DefaultClientOptions then.  We should think this through and see if there's another way.

Reply:
>I don't know that they always can

We have to. Otherwise, exception formatting would sanitize all the headers even if they were marked as safe by the client author.

Reply:
Ok :)